zypp/base/ExternalDataSource.h

```diff
@@ -69,15 +70,29 @@ namespace zypp {
 
       /**
        * Read one line from the input stream.
-       * Returns the line read, including the terminator.
+       * Returns the line read, including the terminator. n
        */
       std::string receiveLine();
 
+      /**
+       * Read one line from the input stream.
+       * Returns the line read, including the terminator. n
+       * The \a timeout value is to be specified in milliseconds.
+       */
+      std::string receiveLine( const int timeout );
```

Comment:
Is there a usecase for negative timeouts or could it be unsigned?

Reply:
timeout == -1 is "no timeout"

Reply:
And -2,....  I'd feel more comfortable with an unsigned timeout_type and an explicit value to be used for "no timeout".
```
    typedef size_t timeout_type;
    static const timeout_type  no_timeout = static_cast<timeout_type>(-1);
```

BTW: There's a stray 'n'  at the end of L79 (also in L73)

Reply:
Done